### ISSUE TITLE
Visual labels are missing for “Select keyboard type”, “Select theme”, and “Select language” controls.:A11y_Aspire website_Welcome to Aspire_Labels or Instructions:

### ISSUE THREAD

**Issue by PoojaNamde** [microsoft/aspire.dev#220]
### GitHub Tags:
#A11yTCS; #A11yMAS; #A11ySev2; #A11yWCAG; #DesktopWeb; #ChromiumEdge; #Win11; #Developer Community Partners; #SH_AspireWebsite_Web_Jan26; #WCAG3.3.2; Labels or Instructions;#ES Chat;

**"Please do not close this bug. This bug should only be closed by TCS, C+AI Accessibility tester after bug verification."**

Try [ES Chat](https://nam06.safelinks.protection.outlook.com/?url=https:%2f%2feng.ms%2fdocs%2fcoreai%2fdevdiv%2fone-engineering-system-1es%2f1es-docs%2fes-chat%2findex&data=05%7c02%7cv-pawanksing%40microsoft.com%7c26cff27270c448f48a0e08de3f569146%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c639017838221175207%7cUnknown%7cTWFpbGZsb3d8eyJFbXB0eU1hcGkiOnRydWUsIlYiOiIwLjAuMDAwMCIsIlAiOiJXaW4zMiIsIkFOIjoiTWFpbCIsIldUIjoyfQ%3D%3D%7c0%7c%7c%7c&sdata=ju8Raw6mIhT5G3gWbZSkS0CAN2zkv2ycA5ifSSqoxOw%3D&reserved=0) to learn more about the MAS rule and how to fix the issue. If you need more help, use our [Teams channel](https://nam06.safelinks.protection.outlook.com/?url=https:%2f%2fteams.microsoft.com%2fl%2fchannel%2f19%253ADeCtYeREyQHw5jYsz9PPc3hpVw48m-tOlTkQ1jEn0eM1%2540thread.tacv2%2fCAI%2520Accessibility%2520Support%2520Channel?groupId%3Df5eb428c-9e6e-4c16-8143-33fe3961a9ae%26tenantId%3D72f988bf-86f1-41af-91ab-2d7cd011db47&data=05%7c02%7cv-pawanksing%40microsoft.com%7c26cff27270c448f48a0e08de3f569146%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c639017838221198042%7cUnknown%7cTWFpbGZsb3d8eyJFbXB0eU1hcGkiOnRydWUsIlYiOiIwLjAuMDAwMCIsIlAiOiJXaW4zMiIsIkFOIjoiTWFpbCIsIldUIjoyfQ%3D%3D%7c0%7c%7c%7c&sdata=DCNVsZe8oDZWqm97AI0ee4OoWWecR6zw6qlhTuzP7MI%3D&reserved=0) or [office hours](https://nam06.safelinks.protection.outlook.com/?url=https:%2f%2foutlook.office.com%2fbook%2fmicrosoft2779007%2540microsoft.onmicrosoft.com%2fs%2fbLeGBltMW0qI0WRLsTqx2A2?ismsaljsauthenabled&data=05%7c02%7cv-pawanksing%40microsoft.com%7c26cff27270c448f48a0e08de3f569146%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c639017838221214666%7cUnknown%7cTWFpbGZsb3d8eyJFbXB0eU1hcGkiOnRydWUsIlYiOiIwLjAuMDAwMCIsIlAiOiJXaW4zMiIsIkFOIjoiTWFpbCIsIldUIjoyfQ%3D%3D%7c0%7c%7c%7c&sdata=CnCCdDgEmll2oUQUdMjXusAgV8PneqZVYvzgxAfk7Ik%3D&reserved=0).

### Environment Details:
Application Name: Aspire website
URL: https://aspire.dev/docs/
Chromium Edge: Version 145.0.3775.0 (Official build) dev (64-bit)
Edition: Windows 11 Enterprise Insider Preview
Version: 24H2
OS build: 26540.1001

### Repro steps:

1. Hit the URL: https://aspire.dev/docs/
2. TAB to “Select keyboard type”, “Select theme”, and “Select language” controls.
3. Observe the issue that Visual labels are missing for “Select keyboard type”, “Select theme”, and “Select language” controls.

### Actual Result:
Visual labels are not provided for “Select keyboard type”, “Select theme”, and “Select language” controls.

### Expected Result:
Visual labels should be provided for “Select keyboard type”, “Select theme”, and “Select language” controls.

### User Impact:
Missing visual labels prevent users with cognitive, low‑vision, or learning disabilities from understanding the purpose of controls, increasing confusion and errors.

### Attachment:

<img width="1366" height="684" alt="Image" src="https://github.com/user-attachments/assets/8d4649b5-77aa-4c03-adf9-cf3c89b8da19" />

**Comment by IEvangelist** [comment]
Hi @PoojaNamde - thank you for this issue, much appreciated. With that said, I wanted a bit of clarification on this. How am I supposed to fix that? Like what specific action needs to be done?

**Comment by PoojaNamde** [comment]
@IEvangelist To fix this issue, programmatically associate descriptive labels with the controls using semantic HTML (e.g., <label> with for attribute) or ARIA attributes like aria-label. This ensures that the labels are accessible to assistive technologies

**Comment by IEvangelist** [comment]
Hi @PoojaNamde - I'm pushing back on this one.

I believe that you're wrong about this being an accessibility issue. There are two ways to associate a `select` control and a corresponding `label`. One way is to (as you mentioned), apply a `for` attribute to the `label` that points to the `select.id`. In this case, the `select` element has an `id`. This is fine for scenarios where labels and selects are next to each other (at the same level) in the DOM.

The other way, (and the way that it's currently implemented) is an implicit associative pattern, in which the `label` literally wraps the `select`. That's the case for all of these controls; this is a non-issue:

> Alternatively, you can nest the <input> directly inside the `<label>`, in which case the `for` and `id` attributes are not needed because the association is implicit.
> - https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/label
